### PR TITLE
Add session stateless filter, #464.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                policyReplyHandler.handle(request, response, chain);
+            } else if (authenticationHandler.matches(request)) {
+                authenticationHandler.handle(request, response, chain);
+            }
+        } catch (AADB2CAuthenticationException e) {
+            throw new ServletException("Validate reply url reqest failure.", e);
+        }
+
+        chain.doFilter(request, response);
+        SecurityContextHolder.getContext().setAuthentication(null);
```
Comment (sophiaso) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilter.java]: Why this `setAuthentication` is after `chain.doFilter(request, response)`?

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilterAuthenticationHandler.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@Slf4j
+public class AADB2CStatelessFilterAuthenticationHandler extends AbstractAADB2CFilterScenarioHandler
+        implements AADB2CFilterScenarioHandler {
+
+    private static final String AUTH_TYPE = "Bearer";
+
+    private static final String AUTH_HEADER = "Authentication";
```
Comment (sophiaso) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilterAuthenticationHandler.java]: The header is `Authentication` but not `Authorization`?

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilterAuthenticationHandler.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public AADB2CStatelessFilterAuthenticationHandler(@NonNull AADB2CProperties b2cProperties) {
+        final String openIdConfigURL = AADB2CURL.getOpenIdSignUpOrInConfigurationURL(b2cProperties);
+
+        this.jwtProcessor = new AADB2CJWTProcessor(openIdConfigURL, b2cProperties);
+    }
+
+    private void handleAuthenticationHeader(HttpServletRequest request, HttpServletResponse response)
+            throws AADB2CAuthenticationException {
+        final String authHeader = request.getHeader(AUTH_HEADER);
+        Assert.isTrue(authHeader.startsWith(AUTH_TYPE), "Unexpected authentication type: " + authHeader);
```
Comment (sophiaso) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilterAuthenticationHandler.java]: Does it sounds better to use TOKEN_TYPE and "Unexpected token type: "?

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilterAuthenticationHandler.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    @Override
+    public void handle(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+            throws AADB2CAuthenticationException {
+        AADB2CURL.validateReplyRequest(request);
+        handleAuthenticationHeader(request, response);
+    }
+
+    @Override
+    public Boolean matches(HttpServletRequest request) {
+        final String authHeader = request.getHeader("Authentication");
```
Comment (sophiaso) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilterAuthenticationHandler.java]: `request.getHeader("Authentication")`, use the constant defined in the beginning?

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilterAuthenticationHandler.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+        this.jwtProcessor = new AADB2CJWTProcessor(openIdConfigURL, b2cProperties);
+    }
+
+    private void handleAuthenticationHeader(HttpServletRequest request, HttpServletResponse response)
+            throws AADB2CAuthenticationException {
+        final String authHeader = request.getHeader(AUTH_HEADER);
+        Assert.isTrue(authHeader.startsWith(AUTH_TYPE), "Unexpected authentication type: " + authHeader);
+
+        final String idToken = authHeader.substring(AUTH_TYPE.length() + 1);
```
Comment (sophiaso) [inline azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilterAuthenticationHandler.java]: The `+1` is used here, how about adding a whitespace after the "Bear" constant?

File: azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilter.java (hunk 1 above)
Comment (Incarnation-p-lee) [reply azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CStatelessFilter.java]: as session stateless, we should only verify token for each request and keep server stateless.

**Pull request merged**